src/uct/ib/base/ib_iface.c

```diff
@@ -614,6 +614,58 @@ ucs_status_t uct_ib_iface_get_device_address(uct_iface_h tl_iface,
     return UCS_OK;
 }
 
+static void uct_ib_iface_log_subnet_info(const struct sockaddr_storage *sa1,
+                                         const struct sockaddr_storage *sa2,
+                                         unsigned prefix_len, int matched)
+{
+    UCS_STRING_BUFFER_ONSTACK(info, UCS_SOCKADDR_STRING_LEN * 2 + 60);
+    char sockstr[UCS_SOCKADDR_STRING_LEN];
+
+    ucs_string_buffer_appendf(&info, "IP addresses");
+    if (!matched) {
+        ucs_string_buffer_appendf(&info, " do not");
+    }
+
+    ucs_string_buffer_appendf(&info, " match with a %u-bit prefix,",
+                              prefix_len);
+
+    ucs_sockaddr_str((struct sockaddr *)sa1, sockstr, UCS_SOCKADDR_STRING_LEN);
```

Comment:
maybe add new string_buffer function, and use it like so:
`ucs_string_buffer_append_saddr(&info, (struct sockaddr *)sa1);`